Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -705,6 +705,8 @@ static String mutationObserverNotificationString()
     auto pageCoverage = pageCoverageForRect(clipRect);
     auto documentScale = pageCoverage.pdfDocumentScale;
 
+    context.clip(clipRect);
```

Comment:
I would do this in the caller. We don't need to clip here in the normal layer painting code path, because the GraphicsLayer code has already applied a clip.

Also, for printing, you should avoid having the tiles paint cached buffers while printing, so that needs wiring up,.

Reply:
Okay, I'll change where we clip.

Reply:
> Also, for printing, you should avoid having the tiles paint cached buffers while printing, so that needs wiring up.

Yes! I'll make a bugzilla for that.